### ISSUE TITLE
bug: ESLint Error

### ISSUE THREAD

**Issue by MattPereira** [scaffold-eth/scaffold-eth-2#901]
### Which method was used to setup Scaffold-ETH 2 ?

git clone, npx create-eth@latest

### Current Behavior

Linter not working so i dont get squigles warning me of unused variables, imports, ect.

![image](https://github.com/user-attachments/assets/0dffd4a5-c00d-431a-8091-c51d9ea48b88)

I do still get blocked from committing and get warnings explained when the lint script runs

![image](https://github.com/user-attachments/assets/830e6025-f605-4648-bf7d-cbb5e632452f)

### Expected Behavior

Here is what it looks like on older version of scaffold when Linter works without error

![image](https://github.com/user-attachments/assets/8df655d2-f66a-4359-b930-54d6cddc5a50)

### Steps To Reproduce

Happens instantly with fresh clone and fresh npx create-eth@latest

### Anything else?

The ESLint error logs

```
[Info  - 1:46:49 PM] ESLint server is starting.
[Info  - 1:46:49 PM] ESLint server running in node v20.9.0
[Info  - 1:46:49 PM] ESLint server is running.
[Info  - 1:46:50 PM] ESLint library loaded from: /home/matthu/Desktop/test/packages/nextjs/node_modules/eslint/lib/api.js
Pages directory cannot be found at /home/matthu/Desktop/test/pages or /home/matthu/Desktop/test/src/pages. If using a custom path, please configure with the `no-html-link-for-pages` rule in your eslint config file.
[Error - 1:46:52 PM] An unexpected error occurred:
[Error - 1:46:52 PM] Error: Cannot find package '@trivago/prettier-plugin-sort-imports' imported from /home/matthu/Desktop/test/noop.js
    at new NodeError (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:16425:5)
    at packageResolve (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:17366:9)
    at moduleResolve (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:17406:18)
    at defaultResolve (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:17497:16)
    at resolve2 (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:17514:12)
    at importFromFile (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:17529:16)
    at importFromDirectory (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:22444:10)
    at importPlugin (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:22456:12)
    at async loadPluginWithoutCache (file:///home/matthu/Desktop/test/packages/nextjs/node_modules/prettier/index.mjs:22460:18)
    at async Promise.all (index 0)
```

**Comment by technophile-04** [comment]
Ohh strange, I tried it with `git clone` and it seems to work for me in vscode and neovim too : 
<details>

<summary>
Neovim : 
</summary>

![Screenshot 2024-07-28 at 12 14 01 PM](https://github.com/user-attachments/assets/99de4f04-549e-4874-8f9d-0b335e3a0907)


</details>

<details>

<summary>
Vscode : 
</summary>

![Screenshot 2024-07-28 at 12 15 16 PM](https://github.com/user-attachments/assets/a3c0b8ec-468f-4403-bdcc-518b41f3835a)

</details>


Maybe something to do with you vscode eslint extension? 

Could you tell more on what extension are using and also which IDE are you using?

**Comment by MattPereira** [comment]
### IDE
- the latest version of Vscode `1.91.1`

### Extensions
- [ES Lint](https://marketplace.visualstudio.com/items?itemName=dbaeumer.vscode-eslint) by microsoft 
- [Error Lense](https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens) 

To get even more weird my `nextjs/.eslintrc.json` file is complaining about the typescript rules, even tho [the file hasn't been touched in 9 months ](https://github.com/scaffold-eth/scaffold-eth-2/blob/main/packages/nextjs/.eslintrc.json):thinking: 

Tried uninstalling to isolate both extensions

### ES Lint
![image](https://github.com/user-attachments/assets/b9352139-7b4d-41d6-b4e6-3768e51f7c81)

### Error Lense
![image](https://github.com/user-attachments/assets/67185870-d937-4a5f-9649-38633277d22e)

**Comment by MattPereira** [comment]
Not using any `.vscode` folder and I try to keep my user `settings.json` light and simple

```json
{
  "workbench.colorTheme": "Atom One Dark",
  "workbench.iconTheme": "vscode-icons",
  "window.titleBarStyle": "custom",
  "editor.fontFamily": "Fira Code",
  "workbench.settings.applyToAllProfiles": ["editor.fontFamily"],
  "editor.defaultFormatter": "esbenp.prettier-vscode",
  "editor.formatOnSave": true,
  "editor.fontLigatures": true,
  "solidity.telemetry": false,
  "github.copilot.enable": {
    "*": true,
    "plaintext": false,
    "markdown": false,
    "scminput": false,
    "solidity": true
  },
  "editor.guides.bracketPairs": true,
  "[solidity]": {
    "editor.defaultFormatter": "NomicFoundation.hardhat-solidity"
  },
  "vsicons.dontShowNewVersionMessage": true,
  "github.copilot.editor.enableAutoCompletions": true,
  "editor.quickSuggestionsDelay": 1,
  "window.zoomLevel": 2,
  "errorLens.enabled": true,
  "window.customTitleBarVisibility": "auto"
}

```

**Comment by MattPereira** [comment]
Prolly not cool enough to switch to make the switch to neovim yet :rofl: 

But very open to using a different extension if there are recommendations

**Comment by rin-st** [comment]
Hey @MattPereira !

Thanks for reporting this!
It doesn't work for me too, but I didn't even notice it 😄 since unused vars have less opacity and it's enough for me

Played with it a bit, and a simple solution that helped me is to remove `"plugin:prettier/recommended"` from `.eslintrc.json` + restart vscode

Possible cause:
- `eslint-plugin-prettier` can [conflict](https://github.com/prettier/eslint-plugin-prettier?tab=readme-ov-file#installation) with other plugins
- our eslint config file already [deprecated](https://eslint.org/docs/latest/use/configure/configuration-files) (see first green block)

I will create PR with eslint config update next days, waiting for `eslint-plugin-prettier` to fix deprecated types.

### ES Lint
> ![image](https://private-user-images.githubusercontent.com/73561520/352870590-b9352139-7b4d-41d6-b4e6-3768e51f7c81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI1MzYwMTcsIm5iZiI6MTcyMjUzNTcxNywicGF0aCI6Ii83MzU2MTUyMC8zNTI4NzA1OTAtYjkzNTIxMzktN2I0ZC00MWQ2LWI0ZTYtMzc2OGU1MWY3YzgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODAxVDE4MDgzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNjMDIxNjJkNjIyZjFhMjA2YjljNGRiZWY2NGUxZjZkY2NiZDBkOTBiMDAxYzNhYTk0NDAwM2FmMDY3ZWYyNDImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.zI5sW9SE_omzgWtzNNypttqGjqrtd9_cpEk7j3CRkZE)

I have this for a long time too, but it's just a warning :). Will fix it

**Comment by MattPereira** [comment]
Hey @rin-st!

Thanks for the assistance! :pray: 

Unfortunately removing `"plugin:prettier/recommended"` from `.eslintrc.json` causes this linter complaint at the top of every file

![image](https://github.com/user-attachments/assets/1988279d-cdbb-40a1-8b38-ec934337e9b6)

And also blocks commits with linter turned on :smiling_face_with_tear: 

![image](https://github.com/user-attachments/assets/d183b268-56e1-4599-a345-9c9bca01daee)

**Comment by MattPereira** [comment]
> It doesn't work for me too, but I didn't even notice it 😄 since unused vars have less opacity and it's enough for me

I think there are other benefits besides just the unused imports / vars. The linter being broken blocks out all warnings including remembering to add dependencies in complex `useEffect` hooks

![image](https://github.com/user-attachments/assets/8081b9ad-5899-40f1-8f44-cc72e9b44ea0)


And if you miss a single warning the linter check will fail in the github action, which is a small annoyance requiring an extra commit to fix :melting_face:

**Comment by rin-st** [comment]
> > It doesn't work for me too, but I didn't even notice it 😄 since unused vars have less opacity and it's enough for me
> 
> I think there are other benefits besides just the unused imports / vars. The linter being broken blocks out all warnings including missing dependencies in complex `useEffect` hooks
> 
> ![image](https://private-user-images.githubusercontent.com/73561520/354715770-8081b9ad-5899-40f1-8f44-cc72e9b44ea0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI4NjA3MTQsIm5iZiI6MTcyMjg2MDQxNCwicGF0aCI6Ii83MzU2MTUyMC8zNTQ3MTU3NzAtODA4MWI5YWQtNTg5OS00MGYxLThmNDQtY2M3MmU5YjQ0ZWEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA1VDEyMjAxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZhMTUzYzc5ZTQ3NzQyYzk0YzZjMmVmMzQ0ODA4ZmM1NTNiMTU0YjA3ZThlNGZlZGI3ZjIyOTUxNWVkYjg3ODEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.k0X8VM9CwglBSW9WeCCXYmAVw99kMody25qafdL0rzY)
> 
> And if you miss a single warning the linter check will fail in the github action, which is a small annoyance requiring an extra commit to fix 🫠

This worked for me, since it's `eslint`, not `typescript-eslint`.

Please check https://github.com/scaffold-eth/scaffold-eth-2/pull/904 . Is it working for you?

**Comment by anton-karlovskiy** [comment]
> > It doesn't work for me too, but I didn't even notice it 😄 since unused vars have less opacity and it's enough for me
> 
> I think there are other benefits besides just the unused imports / vars. The linter being broken blocks out all warnings including missing dependencies in complex `useEffect` hooks
> 
> ![image](https://private-user-images.githubusercontent.com/73561520/354715770-8081b9ad-5899-40f1-8f44-cc72e9b44ea0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjI5NzgxODgsIm5iZiI6MTcyMjk3Nzg4OCwicGF0aCI6Ii83MzU2MTUyMC8zNTQ3MTU3NzAtODA4MWI5YWQtNTg5OS00MGYxLThmNDQtY2M3MmU5YjQ0ZWEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA2VDIwNTgwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTc0MGU1MDM4Mjg2YzExYjk2NjEzZDBiNWJmYzdhNjNhODZiNDM1OTExOGQ1MmQ2MTAyZTFkZjc1MjE1YzVjZWMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.SyrEcKCcH37bDACxD33jvFTOcCLWFfIpqFTxau37WeY)
> 
> And if you miss a single warning the linter check will fail in the github action, which is a small annoyance requiring an extra commit to fix 🫠

I agree!

**Comment by technophile-04** [comment]
> IDE is the latest version of Vscode 1.91.1

Could you try upgrading the version to `1.92.0` ? Also could paste the details of your vs code? You could get them by doing : 

cmd + shift + p and then search for "About", click "Help:About" it should show you an modal with all details. 

<details>

<summary>
Here are my details : 
</summary>

```
Version: 1.92.0 (Universal)
Commit: b1c0a14de1414fcdaa400695b4db1c0799bc3124
Date: 2024-07-31T23:26:45.634Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Darwin arm64 23.5.0
```

</details>

I am using vscode `1.92.0` and it seems to work for me, so not sure how to reproduce it :( 

<details>

<summary>
working ss : 
</summary>

![Screenshot 2024-08-07 at 9 21 09 AM](https://github.com/user-attachments/assets/21ed585d-8de4-4032-bf1b-c657d77ca7f4)


</details>

**Comment by rin-st** [comment]
> Here are my details :

I have same details, but 

<img width="624" alt="image" src="https://github.com/user-attachments/assets/9e522b95-7746-411c-8e4e-081ea3cc6306">

for both node 18/20 (not sure it changes something)

**Comment by anton-karlovskiy** [comment]
> > Here are my details :
> 
> I have same details, but no underlines
> 
> <img alt="image" width="624" src="https://private-user-images.githubusercontent.com/25638585/355728076-9e522b95-7746-411c-8e4e-081ea3cc6306.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwMTgxODYsIm5iZiI6MTcyMzAxNzg4NiwicGF0aCI6Ii8yNTYzODU4NS8zNTU3MjgwNzYtOWU1MjJiOTUtNzc0Ni00MTFjLThlNGUtMDgxZWEzY2M2MzA2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA3VDA4MDQ0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRjZDU2ZTliZmIxNzczNzhmOGI2M2M2OWZkZjcxMzAwZjJhNjE3ZjEwZWQ0YWUzOTk2ZmM3MGM0OGM2NWYyYjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IrJJvHGlzBYKEIdWhF5_uClzykJ50YOWb1thTn8bWo0">
> for both node 18/20 (not sure it changes something)

I have the same details, but no warnings at all.

![image](https://github.com/user-attachments/assets/5a7129ec-03b4-4fed-b443-4701ae4a21c3)

**Comment by portdeveloper** [comment]
`Version: 1.93.0-insider (user setup)
Commit: 39d98602020ddefe53ed217a8c50b9040673a737
Date: 2024-08-05T05:03:45.159Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Windows_NT x64 10.0.22631`


![image](https://github.com/user-attachments/assets/325a3b4b-ce18-4a1d-8b45-068614f424e3)

No warnings on my side as well

**Comment by Pabl0cks** [comment]
Same here. No warnings but same details on hover

> Versión: 1.92.0 (user setup)
Commit: b1c0a14de1414fcdaa400695b4db1c0799bc3124
Date: 2024-07-31T23:26:45.634Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
SO: Windows_NT x64 10.0.19045

Node `v20.11.0`

**Comment by damianmarti** [comment]
No warnings here!

Version: 1.91.1
Commit: f1e16e1e6214d7c44d078b1f0607b2388f29d729
Date: 2024-07-09T22:08:12.169Z
Electron: 29.4.0
ElectronBuildId: 9728852
Chromium: 122.0.6261.156
Node.js: 20.9.0
V8: 12.2.281.27-electron.0
OS: Linux x64 5.4.0-190-generic

Node v18.17.1

**Comment by MattPereira** [comment]
<details><summary>My VS Code Details</summary>

```
Version: 1.92.0
Commit: b1c0a14de1414fcdaa400695b4db1c0799bc3124
Date: 2024-07-31T23:26:45.634Z
Electron: 30.1.2
ElectronBuildId: 9870757
Chromium: 124.0.6367.243
Node.js: 20.14.0
V8: 12.4.254.20-electron.0
OS: Linux x64 6.5.0-45-generic
```

</details>

```
[ pool-creator ] $ node --version
v20.11.1
[ pool-creator ] $ nvm use 18
Now using node v18.18.2 (npm v9.8.1)
```

And still no luck after switching to node v18
![PepeHands](https://github.com/user-attachments/assets/f993aec5-3a25-4086-bc4e-dfd2b1712910)

**Comment by anton-karlovskiy** [comment]
> My VS Code Details
> ```
> Version: 1.92.0
> Commit: b1c0a14de1414fcdaa400695b4db1c0799bc3124
> Date: 2024-07-31T23:26:45.634Z
> Electron: 30.1.2
> ElectronBuildId: 9870757
> Chromium: 124.0.6367.243
> Node.js: 20.14.0
> V8: 12.4.254.20-electron.0
> OS: Linux x64 6.5.0-45-generic
> ```
> 
> ```
> [ pool-creator ] $ node --version
> v20.11.1
> [ pool-creator ] $ nvm use 18
> Now using node v18.18.2 (npm v9.8.1)
> ```
> 
> And still no luck after switching to node v18 and restarting vscode ![PepeHands](https://private-user-images.githubusercontent.com/73561520/355929970-f993aec5-3a25-4086-bc4e-dfd2b1712910.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjMwNTU0NjAsIm5iZiI6MTcyMzA1NTE2MCwicGF0aCI6Ii83MzU2MTUyMC8zNTU5Mjk5NzAtZjk5M2FlYzUtM2EyNS00MDg2LWJjNGUtZGZkMmIxNzEyOTEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODA3VDE4MjYwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhY2YyOGQ3YTNiYzM3MzIwMDgxOGM1ZmUyM2Y0ZTI5NGYyNThhYmFhMWRjMWZiMzViMzVmOGM3ZGExYjMxN2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.doNAS9yAHCQsU1bFHDO8bqTA_Pe5zfaESC5lYRv24bg)

As far as I'm concerned, VS code version or node version does not have to do with this issue. This has to be fixed code-wise.

cc @technophile-04

**Comment by technophile-04** [comment]
Just update on this, was debugging with @MattPereira and seems like #875 is the culprit if he checks out to #869 i.e after clonning the main repo and then doing : 

```
git checkout a78be8761e1d039218cccf57d60a4de56261d859 && yarn install
```

It does seem to solve the problem for him. 

Would love if you guys could test it out. 

Also if it doesn't work then maybe #904 would the way to go it seems

**Comment by technophile-04** [comment]
Thanks all! Closing this since it was solved in #905
